src/hydra/dataProvider.ts

```diff
@@ -604,6 +609,51 @@ function dataProvider(
         ),
       );
 
+  /*
+   * The fetchAllPages method allows running as many requests as needed in order to load all pages of a list.
+   * This function uses the already transformed react-admin response data and could be improved by using the hydra pagination
+   */
+  const fetchAllPages = async (
+    resource: string,
+    params: Partial<GetListParams>,
+    previousResult?: GetListResult,
+  ): Promise<GetManyResult> => {
+    params.pagination = {
+      // Using a default perPage value big enough to limit the number of requests
+      // If the API has a max_per_page value, the perPage parameter can be ignored
+      perPage: params.pagination?.perPage || 100,
+      page: params.pagination?.page || 1,
+    };
+
+    // Get the current page data
+    const pageResult = (await fetchApi(GET_LIST, resource, {
+      ...params,
+      sort: { ...(params.sort || {}) }, // GetListParams requires a full 'sort' object, but it's not mandatory for the Hydra method
+    })) as GetListResult; // As the fetchApi return is not typed for now, assuming that the result is the one we want
+
+    const result = previousResult ?? pageResult;
```

Comment:
Why not:
```ts
const result = [...(previousResult ?? []), ...pageResult];
```

Reply:
previousResult & pageResult are not arrays, but objects with a data & total keys.